dra-evolution/pkg/api/capacity_types.go

```diff
@@ -138,3 +165,29 @@ type DeviceAttribute struct {
 
 const DeviceAttributeMaxIDLength = 32
 const DeviceAttributeMaxValueLength = 64
+
+// SharedCapacity is a set of resources which can be drawn down upon.
+type SharedCapacity struct {
+	// Name is the name of this set of shared capacity, which is unique
+	// resource pool.
+	// +required
+	Name string `json:"name"`
+
+	// Resources are the list of resources provided in this set.
+	Resources []ResourceCapacity `json:"resources,omitempty"`
+}
+
+// SharedCapacityRequest is a request to draw down resources from a particular
+// SharedCapacity.
+type SharedCapacityRequest struct {
+
+	// Name is the name of the SharedCapacity from which to draw the
+	// resources.
+	//
+	// +required
+	Name string `json:"name"`
```

Comment:
Can you show some YAML examples how this is going to be used? How is potential name ambiguity resolved once someone asks for for some kind of device which could be provided by different vendors?

Reply:
My gut feeling is that this may depend on the same naming convention as for device attributes and also shouldn't overlap with those, i.e. "memory" in the `gpu.dra.example.com/` scope can be either an attribute or a consumable resource, but not both.

Reply:
It will be used like this:
https://github.com/kubernetes-sigs/wg-device-management/pull/27#discussion_r1634768662

It is *not* referenceable outside of the devices in this particular slice. Meaning it's not visible to CEL expressions. If you want it to be visible in a CEL expression, you need to duplicate it as an attribute.

Reply:
I think I was confused by the "request" part of the name. This isn't something that users put into a claim to "request" this amount.

Do users ever need to interact with this in a claim? How do they ask for "a GPU with at least 10Gi memory"?

Reply:
Overlapping comments... I think you answered my question. Still, including YAMLs in the PR would help to clarify this.

Reply:
To maybe be more clear -- `sharedCapacities` exists for the scheduler, not the end user. The scheduler will track each `SharedCapacity` and see what's still available in it, when considering a device for allocation that references it in its `SharedCapacityConsumed`. If there are not enough resources available in it (because they have already been consumed by previously allocated devices), then the device is not selectable.

Reply:
I will be sure to include YAMLs in the PR when adding text about this to https://github.com/kubernetes/enhancements/pull/4709

Reply:
No, users never interact with this. It is internal to the device/capacity model. We can use a different word than "Request"...

Reply:
Nothing changes on the claim side for this. It just helps us keep track of which partitions (devices) are allowed and which are already used up, since they can be *indirectly* used up by another device using a shared resource.